homeassistant/components/tradfri/sensor.py

```diff
@@ -27,43 +66,57 @@ async def async_setup_entry(
     coordinator_data = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
     api = coordinator_data[KEY_API]
 
-    async_add_entities(
-        TradfriBatterySensor(
-            device_coordinator,
-            api,
-            gateway_id,
-        )
-        for device_coordinator in coordinator_data[COORDINATOR_LIST]
+    entities: list[TradfriSensor] = []
+
+    for device_coordinator in coordinator_data[COORDINATOR_LIST]:
         if (
             not device_coordinator.device.has_light_control
             and not device_coordinator.device.has_socket_control
             and not device_coordinator.device.has_signal_repeater_control
             and not device_coordinator.device.has_air_purifier_control
-        )
-    )
+        ):
+            entities.append(
+                TradfriSensor(
```

Comment:
Since it's only the description that changes, we could consider storing the description in a local variable in the if blocks and then afterwards add an entity to the list with that description if it's not None. It may save some lines.